libraries/autoload.rb

```diff
@@ -1,3 +1,7 @@
+unless Gem::Requirement.new(">= 12.0").satisfied_by?(Gem::Version.new(Chef::VERSION))
+  raise "The compat_resources cookbook does not support chef versions older than Chef 12.0.0"
```

Comment:
I don't think there's any reason to prevent someone from trying this on 12.0 and it should work, but 12.0.x was just terrible so we don't promise anything.